src/main/java/com/terraformersmc/terrestria/config/TerrestriaConfigManager.java

```diff
@@ -3,140 +3,132 @@
-	public static final Gson GSON = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).setPrettyPrinting().create();
-
-	static {
-		// Make sure the configuration is ready
-		initializeGeneralConfig();
-	}
+	private final Path clientConfigPath;
+	private final Path clientConfigBackupPath;
+	private final Path generalConfigPath;
+	private final Path generalConfigBackupPath;
+	private TerrestriaClientConfig clientConfig;
+	private TerrestriaGeneralConfig generalConfig;
 
-	private static void prepareFiles() {
-		if (directory == null) {
-			directory = new File(FabricLoader.getInstance().getConfigDirectory(), Terrestria.MOD_ID);
-			if (!directory.exists()) {
-				directory.mkdir();
-			}
-		}
-		if (clientFile == null && FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT) {
-			clientFile = new File(directory, "client.json");
-		}
-		if (generalFile == null) {
-			generalFile = new File(directory, "general.json");
-		}
-	}
+	private static final Gson GSON = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).setPrettyPrinting().create();
 
-	public static TerrestriaClientConfig initializeClientConfig() {
-		if (FabricLoader.getInstance().getEnvironmentType() != EnvType.CLIENT) {
-			throw new UnsupportedOperationException("Cannot initialize the Terrestria client configuration when not on the client.");
-		}
-		if (clientConfig != null) {
-			return clientConfig;
-		}
+	public TerrestriaConfigManager() {
+		Path configDirectory = FabricLoader.getInstance().getConfigDir().resolve(Terrestria.MOD_ID);
 
-		clientConfig = new TerrestriaClientConfig();
-		loadClient();
-		return clientConfig;
-	}
-
-	public static TerrestriaGeneralConfig initializeGeneralConfig() {
-		if (generalConfig != null) {
-			return generalConfig;
+		try {
+			Files.createDirectories(configDirectory);
+		} catch (IOException e) {
+			Terrestria.LOGGER.error("Failed to create config directory at " + configDirectory, e);
 		}
 
-		generalConfig = new TerrestriaGeneralConfig();
-		loadGeneral();
+		if (FabricLoader.getInstance().getEnvironmentType() == EnvType.CLIENT) {
+			clientConfigPath = configDirectory.resolve("client.json");
+			clientConfigBackupPath = configDirectory.resolve("client-invalid-syntax.json");
+		} else {
+			clientConfigPath = null;
+			clientConfigBackupPath = null;
+		}
 
-		return generalConfig;
+		generalConfigPath = configDirectory.resolve("general.json");
+		generalConfigBackupPath = configDirectory.resolve("general-invalid-syntax.json");
 	}
 
-	private static void loadClient() {
-		if (FabricLoader.getInstance().getEnvironmentType() != EnvType.CLIENT) {
-			throw new UnsupportedOperationException("Cannot load the Terrestria client configuration when not on the client.");
-		}
-		prepareFiles();
+	private static <T> T loadConfig(Path configPath, T defaults, Class<T> clazz, Path backupPath) {
+		if (!Files.exists(configPath)) {
+			saveConfig(configPath, defaults);
```

Comment:
would be nice to just return the success of saveConfig on the method call instead of having to check if the file exists twice here.